repos/keg-cli/src/libs/git/repo.js

```diff
@@ -64,18 +64,39 @@ class Repo {
   }
 
   /**
-  * Fetches branches from a remote
+  * Fetches all branches from a remote
+  * <br/>Then loop over them and setup the remote branch to be tracked by the local branch
   * @memberof Git
   * @function
   * @param {string} params.location - Location to check
   * @param {Object} cmdOpts - Options to pass to the spawnCmd
   *
   * @returns {void}
   */
-  fetch = ({ location=process.cwd(), env, log, ...params }, cmdOpts) => {
-    cmdOpts = location ? { ...cmdOpts, cwd: location } : cmdOpts
-
-    return gitCmd(`fetch ${ getFetchArgs(params) }`.trim(), cmdOpts, log)
+  fetch = async ({ location=process.cwd(), env, log, ...params }, extraOpts) => {
+    const cmdOpts = location ? { ...extraOpts, cwd: location } : extraOpts
+
+    const response = await gitCmd(`fetch ${ getFetchArgs(params) }`.trim(), cmdOpts, log)
+    // response is the exit code from the git command
+    // So if it is anything other then 0, then there was an error. We just want to return it
+    if(response) return response
+
+    // Get the remote branches to setup tracking
+    const remoteBranches = await this.git.branch.listRemote(location)
+
+    // Loop over each remote branch and set it up to be tracked by the local branch
+    return Promise.all(
+      remoteBranches.map(branch => {
+        try {
+          return gitCli({
+            opts: `branch --track ${branch.name} ${branch.remote}/${branch.name}`,
+            ...options,
```

Comment:
this throws a silent error that `options` is not defined, so i dont think this git cmd is successful on any of the branch

this cmd is suppose to create local branches that I don't have and tie it to the remote right? it is currently not doing that.

Reply:
This is fixed